### PR TITLE
Scheduled Updates: Use Cron API directly

### PR DESCRIPTION
**Pull request by obenland**
Will need to be rebased to trunk, once https://github.com/Automattic/jetpack/pull/36835 made it in.

## Proposed changes:

* Reuses endpoint hooks to clear cache
* Replaces wrapper functions with Cron API calls
* Reuses endpoint methods when updating schedules after a plugin was deleted.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Create a schedule with a few plugins.
* Delete one of the plugins in the schedule.
* Make sure the schedule was updated to remove the plugin.
* Make sure the event logs etc were maintained.

### CODE REVIEW

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php [hunk 1]
```diff
@@ -427,32 +402,29 @@ public static function deleted_plugin( $plugin_file, $deleted ) {
...
-
-			if ( ! count( $plugins ) ) {
-				continue;
-			}
-
-			// There are still plugins to update. Schedule a new event.
-			$result = self::create_scheduled_update( $event->timestamp, $event->schedule, $plugins );
+			if ( empty( $plugins ) ) {
+				// Delete the schedule if there are no more plugins.
+				$request = new \WP_REST_Request( 'PUT', '/wpcom/v2/update-schedules/' . $id );
```
Comment (ouikhuan) [inline projects/packages/scheduled-updates/src/class-scheduled-updates.php]: Should the method be `DELETE` here instead?

File: projects/packages/scheduled-updates/src/class-scheduled-updates.php [hunk 2]
```diff
@@ -427,32 +402,29 @@ public static function deleted_plugin( $plugin_file, $deleted ) {
...
 
-			if ( is_wp_error( $result ) || false === $result ) {
-				continue;
-			}
-
-			$schedule_id = self::generate_schedule_id( $plugins );
-			$status      = self::get_scheduled_update_status( $id );
+				$endpoint->delete_item( $request );
+			} else {
+				$request = new \WP_REST_Request( 'DELETE', '/wpcom/v2/update-schedules/' . $id );
```
Comment (ouikhuan) [inline projects/packages/scheduled-updates/src/class-scheduled-updates.php]: Should the method be `PUT` here instead?

**Comment by obenland** [comment]
This still might not work as expected, as it is now. See p1713887825967409/1713886953.545839-slack-C01A60HCGUA

Still have to test this a bit more I think. So far I've not even been able to reproduce the original bug with outdated caches

**Review by ouikhuan — APPROVED** [review]
> This still might not work as expected, as it is now. See p1713887825967409/1713886953.545839-slack-C01A60HCGUA
> 
> Still have to test this a bit more I think. So far I've not even been able to reproduce the original bug with outdated caches

Agreed, I've tested the current implementation several times but unable to reproduce the cache issue either. I also echo that if we want to do this we will need to do it on every cron update, not just when we delete a schedule because the original bugs was happening across all the CRUD actions. I'll approve the PR for now in case you want to iterate it later.

**Comment by obenland** [comment]
I don't think I'll get a chance to work on that again any time soon

**Pull request closed without merging**